ci: oduran/xml_validation

OmarDuran opened the pull request:
Simplified Validation Bash Script

Proposed Changes:
This PR simplifies the validation bash script along with some minor improvements on documentation for its usage.

Considerations:
- Calls to Git for inferring a list of XML files for validation have been removed.
- The Git repository directory is now explicitly passed as input to the validation script.
- The script specifically validates files located in the 'examples' and 'inputfiles' directories.
- Python scripting is discarded as an option, as relying on a specific Python version for the script is not ideal.

Outcome:

- A simpler bash script that is easier to maintain and use.
- The process will no longer fail silently; if a file doesn't validate, the error will be clearly reported.

OmarDuran committed:
wip: simplify validation scheme

wip: simplify validation scheme

wip: reactivating xml validation

wip: test

Update validateXMLFiles.bash

Update build_and_test.yml

Update validateXMLFiles.bash

wip: simple test

wip: test

wip: run container as a root

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip: improving logic

wip:test

wip: test

wip: fix

wip: test

wip:test

wip: test

wip: test

wip: test

wip: test

wip: test

wip: test

wip: test

wip: test

wip: test

wip: fix

wip: test

wip: passing only ${CMAKE_SOURCE_DIR}

wip: level up

wip: shift input path

wip: adding extra command

wip: theirs

wip: theirs

wip: emit location after cd

wip: print git version

wip: performing cd path externally

wip: use find at given path

wip: removing git commands

wip: improving search with find

wip: check schema location

wip: entering

Update validateXMLFiles.bash

wip: simplify validation scheme

Update validateXMLFiles.bash

Update validateXMLFiles.bash

wip: run as nobody user but in the root group

wip: run file by file

wip: no-silent failure

OmarDuran committed:
wip: simplify validation scheme

wip: reactivating xml validation

wip: test

Update validateXMLFiles.bash

Update build_and_test.yml

Update validateXMLFiles.bash

wip: simple test

wip: test

wip: run container as a root

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip:test

wip: improving logic

wip:test

wip: test

wip: fix

wip: test

wip:test

wip: test

wip: test

wip: test

wip: test

wip: test

wip: test

wip: test

wip: test

wip: test

wip: fix

wip: test

wip: passing only ${CMAKE_SOURCE_DIR}

wip: level up

wip: shift input path

wip: adding extra command

wip: theirs

wip: theirs

wip: emit location after cd

wip: print git version

wip: performing cd path externally

wip: use find at given path

wip: removing git commands

wip: improving search with find

wip: check schema location

wip: entering

Update validateXMLFiles.bash

wip: simplify validation scheme

Update validateXMLFiles.bash

OmarDuran committed:
Update validateXMLFiles.bash

OmarDuran committed:
wip: run as nobody user but in the root group

OmarDuran committed:
wip: run file by file

OmarDuran committed:
wip: no-silent failure

OmarDuran committed:
wip: fix files that does not validate

paveltomin commented on inputFiles/compositionalMultiphaseFlow/overallCompositionFormulation/co2_2d_plume/co2_2d_plume_Z.xml:
```diff
@@ -1,14 +1,12 @@
       logLevel="1"
       discretization="fluidTPFA"
       temperature="368.15"
-      formulationType="OverallComposition"
```
not sure why this got changed

OmarDuran replied:
See reply below.

paveltomin commented on inputFiles/compositionalMultiphaseFlow/overallCompositionFormulation/gravitySegregation/grav_seg_drain_Z.xml:
```diff
@@ -14,8 +12,7 @@
       discretization="fluidTPFA"
       temperature="300"
       initialDt="10"
-      formulationType="OverallComposition"
```
`OverallComposition` should be totally fine to use

OmarDuran replied:
By mistake, I changed the XML files while trying to make them compliant with an outdated schema. The schema has since been updated, and therefore the XML files have been restored to their original settings.

OmarDuran committed:
Merge branch 'develop' into ci/oduran/xml_validation

OmarDuran committed:
wip: removing realpath: illegal option -- s

OmarDuran committed:
Merge branch 'ci/oduran/xml_validation' of https://github.com/GEOS-DEV/GEOS into ci/oduran/xml_validation

OmarDuran committed:
Merge branch 'develop' into ci/oduran/xml_validation

OmarDuran committed:
wip: update schema file

OmarDuran committed:
Merge branch 'develop' into ci/oduran/xml_validation

OmarDuran committed:
wip: remove print

wip: shorter variable name

Update grav_seg_drain_Z.xml

Update validateXMLFiles.bash

OmarDuran commented:
Merging is blocked: Waiting on code owner review from @dkachuma, @paveltomin , and/or @tjb-ltk.

OmarDuran committed:
Merge branch 'develop' into ci/oduran/xml_validation

paveltomin committed:
Update grav_seg_drain_Z.xml
